browser/ui/webui/brave_rewards/rewards_page_handler.cc

```diff
@@ -182,10 +236,146 @@ void RewardsPageHandler::GetExternalWalletProviders(
+  settings->notification_ads_enabled =
+      prefs_->GetBoolean(brave_ads::prefs::kOptedInToNotificationAds);
+  settings->search_ads_enabled =
+      prefs_->GetBoolean(brave_ads::prefs::kOptedInToSearchResultAds);
+  settings->inline_content_ads_enabled =
+      prefs_->GetBoolean(brave_news::prefs::kBraveNewsOptedIn) &&
+      prefs_->GetBoolean(brave_news::prefs::kNewTabPageShowToday);
+
+  settings->notification_ads_per_hour =
+      ads_service_->GetMaximumNotificationAdsPerHour();
+
+  settings->current_subdivision = prefs_->GetString(
+      brave_ads::prefs::kSubdivisionTargetingUserSelectedSubdivision);
+  settings->auto_detected_subdivision = prefs_->GetString(
+      brave_ads::prefs::kSubdivisionTargetingAutoDetectedSubdivision);
+  settings->should_allow_subdivision_targeting =
+      prefs_->GetBoolean(brave_ads::prefs::kShouldAllowSubdivisionTargeting);
+
+  auto& subdivisions = brave_ads::GetSupportedSubdivisions();
+  auto iter = subdivisions.find(brave_l10n::GetCountryCode(GetLocalState()));
+  if (iter != subdivisions.cend()) {
+    for (auto& [code, name] : iter->second) {
+      auto entry = mojom::AdsSubdivision::New();
+      entry->code = code;
+      entry->name = name;
+      settings->available_subdivisions.push_back(std::move(entry));
+    }
+  }
+
+  std::move(callback).Run(std::move(settings));
+}
+
 void RewardsPageHandler::GetAdsStatement(GetAdsStatementCallback callback) {
   ads_service_->GetStatementOfAccounts(std::move(callback));
 }
 
+void RewardsPageHandler::GetAdsHistory(GetAdsHistoryCallback callback) {
+  base::Time now = base::Time::Now();
+  base::Time from_time =
+      now - brave_ads::kAdHistoryRetentionPeriod - base::Days(1);
+
+  auto on_history = [](decltype(callback) callback, base::Value::List list) {
+    // The Ads service provides Ads history data as a `base::Value` (i.e. JSON).
+    // Rather than sending a Mojo `base::Value` interface to the client (which
+    // is awkward to use in this context), send the data to the WebUI as a JSON
+    // string. The front-end will send this JSON data back when the user
+    // modifies Ads history state.
+    std::string json;
+    if (!base::JSONWriter::Write(list, &json)) {
+      LOG(ERROR) << "Unable to convert Ads history to JSON";
+    }
+    std::move(callback).Run(std::move(json));
+  };
+
+  ads_service_->GetAdHistory(from_time.LocalMidnight(), now,
+                             base::BindOnce(on_history, std::move(callback)));
+}
+
+void RewardsPageHandler::SetAdTypeEnabled(mojom::AdType ad_type,
+                                          bool enabled,
+                                          SetAdTypeEnabledCallback callback) {
+  switch (ad_type) {
+    case mojom::AdType::kNewTabPage:
+      prefs_->SetBoolean(ntp_background_images::prefs::
+                             kNewTabPageShowSponsoredImagesBackgroundImage,
+                         enabled);
+      break;
```

Comment:
In the ads component we check both `kNewTabPageShowSponsoredImagesBackgroundImage` and `kNewTabPageShowBackgroundImage` are true, if we only set this pref ads may not be shown.

Reply:
Good catch. This code is copying the current Rewards page behavior. @Miyayes what do you think? Should we change the behavior of the "New Tab Page Ads" toggle so that it controls both prefs? Or should that be a separate issue?